pkg/controller/job-controller.go

```diff
@@ -203,16 +285,428 @@ func (c *JobController) SyncHandler(key string) (Result, error) {
+			// if success is false and message is empty, it means the job is running
+			if command.Message == "" {
+				running = true
+				status.CommandsStatus = append(status.CommandsStatus, v1alpha1.CommandStatus{
+					Name:    command.JobName,
+					Result:  "running",
+					Message: command.Message,
+				})
+			} else {
+				status.CommandsStatus = append(status.CommandsStatus, v1alpha1.CommandStatus{
+					Name:    command.JobName,
+					Result:  "failed",
+					Message: command.Message,
+				})
+			}
+		}
+		command.Locker.RUnlock()
+	}
+	if running {
+		status.Phase = MinioJobPhaseRunning
+	} else {
+		if failed {
+			status.Phase = MinioJobPhaseFailed
+			status.Message = message
+		} else {
+			status.Phase = MinioJobPhaseSuccess
+		}
+	}
+	return status
+}
+
+func (intervalJob *MinIOIntervalJob) createCommandJob(ctx context.Context, k8sClient client.Client) error {
+	for _, command := range intervalJob.Command {
+		if len(command.DepnedOn) == 0 {
+			err := command.createJob(ctx, k8sClient, intervalJob.JobCR)
+			if err != nil {
+				return err
+			}
+		} else {
+			dependAllSucess := true
+			for _, dep := range command.DepnedOn {
+				status, found := intervalJob.CommandMap[dep]
+				if !found {
+					return fmt.Errorf("depended job %s not found", dep)
+				}
+				if !status.success() {
+					dependAllSucess = false
+					break
+				}
+			}
+			if dependAllSucess {
+				err := command.createJob(ctx, k8sClient, intervalJob.JobCR)
+				if err != nil {
+					return err
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func checkMinIOJob(jobCR *v1alpha1.MinIOJob) (intervalJob *MinIOIntervalJob, err error) {
+	defer func() {
+		if err != nil {
+			globalIntervalJobStatus.Delete(fmt.Sprintf("%s/%s", jobCR.Namespace, jobCR.Name))
+		}
+	}()
+	val, found := globalIntervalJobStatus.Load(fmt.Sprintf("%s/%s", jobCR.Namespace, jobCR.Name))
+	if found {
+		intervalJob = val.(*MinIOIntervalJob)
+		if reflect.DeepEqual(intervalJob.JobCR.Spec, jobCR.Spec) {
+			intervalJob.JobCR.UID = jobCR.UID
+			return intervalJob, nil
+		}
+	}
+	intervalJob = &MinIOIntervalJob{
+		JobCR:      jobCR.DeepCopy(),
+		Command:    []*MinIOIntervalJobCommand{},
+		CommandMap: map[string]*MinIOIntervalJobCommand{},
+	}
+	if jobCR.Spec.TenantRef.Namespace == "" {
+		return intervalJob, fmt.Errorf("tenant namespace is empty")
+	}
+	if jobCR.Spec.TenantRef.Name == "" {
+		return intervalJob, fmt.Errorf("tenant name is empty")
+	}
+	if jobCR.Spec.ServiceAccountName == "" {
+		return intervalJob, fmt.Errorf("serviceaccount name is empty")
+	}
+	for index, val := range jobCR.Spec.Commands {
+		mcCommand, found := operationAliasToMC(val.Operation)
+		if !found {
+			return intervalJob, fmt.Errorf("operation %s is not supported", val.Operation)
+		}
+		var command string
+		argsTpl, found := jobOperation[mcCommand]
+		if !found {
+			return intervalJob, fmt.Errorf("operation %s is not supported", mcCommand)
+		}
+		// [ALIAS]
+		command = strings.ReplaceAll(argsTpl, "[ALIAS]", "myminio")
```

Comment:
Should we use template package here?

Reply:
I have another idea for this. But you can test this first.